libraries/chain/backing_store/tests/benchmark_kv.cpp

```diff
@@ -0,0 +1,337 @@
+#include <sys/resource.h>
+
+#include <eosio/chain/kv_chainbase_objects.hpp>
+#include <eosio/chain/backing_store/kv_context_rocksdb.hpp>
+#include <eosio/chain/backing_store/kv_context_chainbase.hpp>
+
+#include <boost/filesystem.hpp>
+
+using namespace eosio;
+using namespace eosio::chain;
+
+namespace kv_benchmark {
+
+// Global test data
+constexpr account_name receiver = N(kvrdb);
+constexpr uint64_t contract = receiver.to_uint64_t();
+constexpr account_name payer = N(payer);
+std::string prefix = "prefix";
+constexpr uint32_t it_key_size = 10;
+constexpr uint64_t default_billable_size = 12;
+
+constexpr uint32_t default_value_size = 1024;
+std::string default_operation = "get";
+struct cmd_args {
+   uint32_t num_keys = 0;
+   uint32_t value_size = default_value_size;
+   bool is_rocksdb = true; // false means chainbase
+   std::string operation = default_operation;
+   std::string key_file = "";
+};
+
+constexpr uint32_t work_set_size = 10000;
+constexpr uint32_t num_iterations = 100;
+
+struct dummy_control {
+   fc::logger* get_deep_mind_logger() {
+      return nullptr;
+   }
+
+   uint32_t get_action_id() {
+      return 0;
+   }
+};
+
+struct dummy_context {
+   dummy_control control;
+
+   uint32_t get_action_id() {
+      return 0;
+   }
+};
+
+struct mock_resource_manager {
+   uint64_t billable_size = default_billable_size;
+
+   int64_t update_table_usage(account_name payer, int64_t delta, const kv_resource_trace& trace) {
+      return 0;
+   }
+
+   dummy_context* _context;
+};
+
+using wall_clock_t = std::chrono::time_point<std::chrono::steady_clock>;
+
+// resource_manager, receiver, limits in kv_context are 
+// initialzed as a reference.  Place them globally to avoid 
+// going out of scope
+constexpr uint32_t max_key_size   = 100;
+constexpr uint32_t max_value_size = 1020*1024;
+constexpr uint32_t max_iterators  = 3;
+kv_database_config limits { max_key_size, max_value_size, max_iterators };
+mock_resource_manager resource_manager;
+
+// Main purpose of this program is to microbenchmark individual
+// KV API over RocksDB and ChainKV.
+
+void print_usage(char* program_name) {
+   std::cout << program_name << " - Microbenchmark KV APIs over RocksDb and Chainbase\n\n"
+      << "   -f=FILENAME, file storing keys one per line \n"
+      << "   -o=OPERATION, operation to be benchmarked: get, get_data, set, it_next, it_key_value, Default " << default_operation << std::endl
+      << "   -c, use Chainbase. Without this option: use RocksDb\n"
+      << "   -h, print this help\n"
+      << "   -v=NUMBER, value size. Default " << default_value_size << std::endl
+      << std::endl;
+}
+
+// As this program is only for internal use,
+// we don't use sophisticated library to parse command line
```

Comment:
Hmm.  Using a library is supposed to be easier.

Reply:
Thanks. I will change to use the command line parsing library.